### PR TITLE
[SPARK-25508][SQL][TEST] Refactor OrcReadBenchmark to use main method

### PR DESCRIPTION
**Pull request by yucai**
## What changes were proposed in this pull request?

Refactor OrcReadBenchmark to use main method.
Generate benchmark result:
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "hive/test:runMain org.apache.spark.sql.hive.orc.OrcReadBenchmark"
```
## How was this patch tested?

manual tests

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala [hunk 1]
```diff
@@ -436,49 +331,36 @@ object OrcReadBenchmark extends SQLHelper {
...
         benchmark.run()
       }
     }
   }
 
-  def main(args: Array[String]): Unit = {
-    Seq(ByteType, ShortType, IntegerType, LongType, FloatType, DoubleType).foreach { dataType =>
-      numericScanBenchmark(1024 * 1024 * 15, dataType)
+  override def benchmark(): Unit = {
+    runBenchmark("SQL Single Column Scan") {
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala]: nit `SQL Single Column Scan` -> `SQL Single Numeric Column Scan`?

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala]: Also, please add `[TEST]` into the title.

**Review by dongjoon-hyun — COMMENTED** [review]
@yucai . Could you review and merge https://github.com/yucai/spark/pull/5 ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**